.github/workflows/deploy.yml

```diff
@@ -0,0 +1,30 @@
+name: Deploy to Heroku
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+    branches:
+      - main
```

Comment:
We should allow none-main branch to be deployed to staging. The use case for this is if we are developing code on our branch and would like Nadia to check out the feature, we should have the option to deploy our experimental code to staging.

Reply:
Yeah, we can still do this--all we'd have to do is have part of our branch changes include changing this line to the name of the branch. Once this is merged, I can show you what I mean (this option doesn't become available until the workflow is first merged to main).
Having the branch specified as main seems to me to be a good way of letting contributors know that only the main branch ought to be manually deployed to prod, while still making it possible to do other types of deployment for development purposes.
As a side note, I checked on the extra logic to ensure only main gets deployed to prod, and that doesn't seem to get us any further, since that can also be changed on a branch.

Reply:
I guess one other way we could do it is to put no up-front restriction on the branch and environment selected by the user, but throw an error and quit the process if the user tries to deploy to prod from a non-main branch. That's fine with me too, though personally, if I were developing with a goal of putting something out there for a user to try, I'd probably prefer to change line 6 to the name of my branch so every push to the branch triggered a deployment. I'll go with whatever you prefer.

Reply:
> Yeah, we can still do this--all we'd have to do is have part of our branch changes include changing this line to the name of the branch. Once this is merged, I can show you what I mean (this option doesn't become available until the workflow is first merged to main).

Do you mean we have to change the `deployment.yaml` on our branch to deploy to staging? Then we would have to remember to change it back before we merge to main. As a reviewer, I don't want to keep track changes in deployment.yaml. When I see changes in deployment.yaml, that tells me we are going to bypass the process we established and raises a red flag for me to look closer.

I may be understanding your statement wrong. I am going to approve the PR so that you can show me.

Reply:
I believe you understand it correctly, meaning we should change it, but I will go ahead and merge because it is easier to experiment with workflows after they have first been merged to main, and I will make a followup PR